internal/subscriber/retry/delayconsumer.go

```diff
@@ -80,6 +82,7 @@ func (dc *DelayConsumer) Run(ctx context.Context) {
 			if err != nil {
 				if !messagebroker.IsErrorRecoverable(err) {
 					logger.Ctx(dc.ctx).Errorw("delay-consumer: error in receiving messages", dc.LogFields("error", err.Error())...)
+					dc.errChan <- err
```

Comment:
This PR aims to restart subscribers on any and every unrecoverable error. Ideally, we want to restart the failed component if possible in the future. Keep this in mind when you plan for the subscriber re-arch